devito/__init__.py

```diff
@@ -45,10 +48,22 @@ def _cast_and_update_compiler(val):
 configuration.add('platform', 'intel64', PLATFORMs)
 
 
+# In develop-mode:
+# - Some optimizations may not be applied to the generated code.
+# - The compiler performs more type and value checking
+def _switch_cpu(develop_mode):
+    if bool(develop_mode) is False:
+        isa, platform = infer_cpu()
+        configuration['isa'] = os.environ.get('DEVITO_ISA', isa)
+        configuration['platform'] = os.environ.get('DEVITO_PLATFORM', platform)
+    else:
+        configuration['isa'] = 'cpp'
+        configuration['platform'] = 'intel64'
+configuration.add('develop-mode', True, [False, True], _switch_cpu)  # noqa
```

Comment:
> Do you want it to default to develop mode?

at this stage, yes

> If so, you might want to throw a warning if the user specifies isa or platform, since you'll be ignoring them, and I think that may lead to some confusion.

I'm planning a PR to improve this kind of things. Note that this is the same behaviour as current master, I just moved this piece of code from yask to the main __init__